### ISSUE TITLE
CREATE USER: exitcode of 'cannot insert' exception is different than what is printed to user

### ISSUE THREAD

**Issue by ritaank** [ClickHouse/ClickHouse#12510]
**Describe the bug**
Whenever facing a 'cannot insert' error—for example, by creating user that already exists (without passing if not exists)—an error is given which has the rough format of the following:

```Code: 493. DB::Exception: Received from localhost:9000. DB::Exception: {type} `{name}`: cannot insert because {type} `{name}` already exists in [disk].```

This type of error makes sense. However, when testing, it seems that the actual underlying exitcode might not be 493, and instead is 237.

**How to reproduce**
```
ClickHouse client version 20.6.1.4066 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.6.1 revision 54436.

clicktest :) create user user0
CREATE USER user0
Ok.
0 rows in set. Elapsed: 0.001 sec. 

clicktest :) create user user0
CREATE USER user0

Received exception from server (version 20.6.1):
Code: 493. DB::Exception: Received from localhost:9000. DB::Exception: user `user0`: cannot insert because user `user0` already exists in [disk]. 
0 rows in set. Elapsed: 0.003 sec. 
```

We expect this above error. We see that the actual exitcode, however, is 237:

```
⟥  When I create row policy policy0
    ⟥  By executing command
                                 echo -e "CREATE USER user0" | clickhouse client -n
                                [clickhouse1] bash# echo -e "CREATE USER user0" | clickhouse client -n
                                [clickhouse1] Received exception from server (version 20.6.1):
                                [clickhouse1] Code: 493. DB::Exception: Received from localhost:9000. DB::Exception: user `user0`: cannot insert because user `user0` already exists in [disk].
                               [clickhouse1] bash# echo $?
                               [clickhouse1] 237
                               [clickhouse1] bash#
    ⟥⟤ OK executing command, /rbac/syntax/create user/I create user that already exists, throws exception/I create user user0/executing command
    ⟥  Then exitcode should be 493
    ⟥    Exception: Traceback (most recent call last):
                                        File "/some/file/location", line 112, in query
                                            assert r.exitcode == exitcode, error(r.output)
                                        AssertionError: Oops! Assertion failed

                                        The following assertion was not satisfied
                                          assert r.exitcode == exitcode, error(r.output)

                                        Description
                                          Received exception from server (version 20.6.1):
                                        Code: 493. DB::Exception: Received from localhost:9000. DB::Exception: user `user0`: cannot insert because user `user0` already exists in [disk].

                                        Assertion values
                                          assert r.exitcode == exitcode, error(r.output)
                                                 ^ is <testflows.connect.shell.Command object at 0x7f519445ad90>
                                          assert r.exitcode == exitcode, error(r.output)
                                                 ^ is = 237
                                          assert r.exitcode == exitcode, error(r.output)
                                                               ^ is 493
                                          assert r.exitcode == exitcode, error(r.output)
                                                            ^ is = False
                                          assert r.exitcode == exitcode, error(r.output)
                                          ^ is False
```
**Expected behavior**
I would expect the printed exit code and the actual exit code to be the same: either both 493, or both 237.
This seems to be consistent among all 'cannot insert' errors of this kind, which occur when creating a {user, row policy, quota, etc.} that already exists.

**Comment by ritaank** [comment]
When using `CREATE ROLE` in a similar situation, the 'actual' exitcode seems to be `0`, not `237` as shown in the `CREATE USER` example above. It seems that this exitcode of `0` is unique to `CREATE ROLE`. Details below.

```
ClickHouse client version 20.6.1.4066 (official build).
Connecting to localhost:9000 as user default.
Connected to ClickHouse server version 20.6.1 revision 54436.

clicktest :) create role role0
CREATE ROLE role0
Ok.
0 rows in set. Elapsed: 0.001 sec. 

clicktest :) create role role0
CREATE ROLE role0

Received exception from server (version 20.6.1):
Code: 493. DB::Exception: Received from localhost:9000. DB::Exception: role `role0`: cannot insert because role `role0` already exists in [disk]. 
0 rows in set. Elapsed: 0.002 sec. 
```
And while testing the actual exitcode:

```
assert r.exitcode == exitcode, error(r.output)
                                        AssertionError: Oops! Assertion failed

                                        The following assertion was not satisfied
                                          assert r.exitcode == exitcode, error(r.output)

                                        Assertion values
                                          assert r.exitcode == exitcode, error(r.output)
                                                 ^ is <testflows.connect.shell.Command object at 0x7f12cc2f9b20>
                                          assert r.exitcode == exitcode, error(r.output)
                                                 ^ is = 0
                                          assert r.exitcode == exitcode, error(r.output)
                                                               ^ is 493
                                          assert r.exitcode == exitcode, error(r.output)
                                                            ^ is = False
                                          assert r.exitcode == exitcode, error(r.output)
                                          ^ is False

```

**Comment by ritaank** [comment]
The posted `CANNOT INSERT` issue seems to be a wraparound case as resolved in #11581.

However, this [comment](https://github.com/ClickHouse/ClickHouse/issues/11581#issuecomment-642618638) from that same issue says that returning exitcode 0 is a bug. This happens in the `CREATE ROLE` comment above.

**Comment by filimonov** [comment]
Exit codes for application is 1 byte integer. It can't go higher that 255.

Real exception codes can be any number.

So it can't be fixed.

We can think of mapping group of exceptions codes to single exit code (we do smth similar for http)
